extension/js/common/settings.ts

```diff
@@ -109,11 +113,14 @@ export class Settings {
     }
   }
 
-  static refreshAcctAliases = async (acctEmail: string) => {
-    const addresses = await Settings.fetchAcctAliasesFromGmail(acctEmail);
-    const all = Value.arr.unique(addresses.concat(acctEmail));
-    await Store.setAcct(acctEmail, { addresses: all });
-    return all;
+  static refreshAcctAliases = async (acctEmail: string): Promise<boolean> => {
+    const fetchedSendAs = await Settings.fetchAcctAliasesFromGmail(acctEmail);
+    const { sendAs: storedAliases } = (await Store.getAcct(acctEmail, ['sendAs']));
+    if (!storedAliases || !compareObjects(fetchedSendAs, storedAliases)) {
```

Comment:
Too complicated, I think unnecessarily.

The only reason we actually need to refresh a window is if there is an alias missing. That means you can just compare:

```js
!storedAliases || Object.keys(fetchedSendAs).sort().join(',') !== Object.keys(storedAliases).sort().join(',')
```

Then you update the stored object either way, but only return true if list of aliases changed.

Reply:
Unfortunately, I also was thinking about it but the reason I had to create this `compareObjects` function is that we can't change default alias using Flowcrpyt now so I need to check not only if alias emails changed but if `isDefault` property is changed too. 

The first idea I came up was to replace `Object.keys`  with `JSON.stringify` but that didn't work because Gmail returned **the same** object with the same properties but in another order so the result of `JSON.stringify` was different. Then I decided to create this function called `compareObjects`.

Reply:
I also agree that this method (`compareObjects`) is too large and complicated but I guess that just checking those two fields (`key` and `isDefault`) would be a little weird and incorrect. So I wrote this general solution.

Reply:
Yes, this is much easier and readable, I thought that this was would be little incorrect but if you wrote this I'm going to replace my code with yours. 
As I said, I wanted to make more general solution but when you show me your example I realized that it seems like it doesn't really make sense

Thanks, I will change it

Reply:
Your solution was a prudent one, to offer best user experience. But it came out as so much work (and so much code) that the tradeoff may not be completely worth it in this case, considering it gets updated next time user reopens the window.